things/policies/service.go

```diff
@@ -285,3 +297,31 @@ func (svc service) usersAuthorize(ctx context.Context, subject, object, action,
 	}
 	return nil
 }
+
+// kv is used to create a key-value pair for caching.
+// If thingID is not empty, it will be appended to the value.
+func kv(p Policy, thingID string) (string, string) {
+	if thingID != "" {
+		return p.Subject + kvSeparator + p.Object, strings.Join(p.Actions, kvSeparator) + kvSeparator + thingID
+	}
+
+	return p.Subject + kvSeparator + p.Object, strings.Join(p.Actions, kvSeparator)
+}
+
+// separateActions is used to separate the actions from the cache values.
+func separateActions(actions string) []string {
+	return strings.Split(actions, kvSeparator)
+}
+
+// extractThingID is used to extract the thingID from the cache values.
+func extractThingID(actions string) string {
+	var lastIdx = strings.LastIndex(actions, kvSeparator)
+
```

Comment:
Move this back to the `redis` implementation, even at the cost of creating an additional structure in the `redis` pkg used as a DTO to avoid circular imports.